### DISCUSSION TITLE
proposal: Databend Platform Support Policy

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#3597] [Ideas]
## Background

Databend used to say that we support running on Linux & macOS both on amd64 & arm64 platforms. But we didn't say we which versions we support.

For example,

- Is databend-qeury works on Ubuntu 16.04 (amd64)?
- Is bendctl works on macOS Mojave (arm64)?

This problem becomes more serious for the Linux platform, as described in those issues:

- https://github.com/datafuselabs/databend/issues/3423
- https://github.com/datafuselabs/databend/issues/2063
- https://github.com/datafuselabs/databend/issues/3570

So, we need to define a clear policy for databend platform support. With this policy, our users will know which platform is native, and which platform needs extra work or cost.

It's a contract between developers and users.

## Proposal

So I propose a draft for our first version of platform support.

- All our binary will share the same policy
  - For now, they are including `databend-query`, `databend-meta`,  `databend-benchmark` and `bendctl`.
- All our binary will support platforms that listed the following:
  - All Ubuntu versions that have standard support (in short, 18.04 and 20.04, https://wiki.ubuntu.com/Releases), both on amd64 and arm64
  - All maintained macOS versions (in short, 10.15, 11, 12, https://computing.cs.cmu.edu/desktop/os-lifecycle), both on amd64 and arm64.
- Failed to run on the supported platform will be recognized as a bug, we will fix them as soon as possible.
- Failed to run on the non-supported platform will be recognized as a feature request, we may provide alternative solutions like
  - statically linked binary (drawback: binary size increased)
  - run as a container (drawback: dependent on docker/podman)

---

Welcome any comments!

**Comment by everpcpc** [comment]
Maybe we should temporarily keep back the arm64 image base?
It seems https://github.com/cross-rs/cross/pull/591 has no progress yet recently.
